rmgpy/molecule/resonance.py

```diff
@@ -482,14 +484,52 @@ def generateAromaticResonanceStructures(mol, features=None):
     for mol0, aromaticBonds in molList:
         if not aromaticBonds:
             continue
+        # Save original bond orders in case this doesn't work out
+        originalBonds = []
+        for ring in aromaticBonds:
+            originalOrder = []
+            for bond in ring:
+                originalOrder.append(bond.order)
+            originalBonds.append(originalOrder)
+        # Change bond types to benzene bonds for all aromatic rings
         for ring in aromaticBonds:
             for bond in ring:
                 bond.order = 1.5
 
         try:
             mol0.updateAtomTypes(logSpecies=False)
         except AtomTypeError:
-            continue
+            # If this didn't work the first time, then there might be a ring that is not actually aromatic
```

Comment:
The last two tests in 7cc13aa6f2e8edcbd98d561554d8c3e8ec4faf27 are cases where the first `updateAtomTypes()` call fails with an `AtomTypeError`, and are designed to test what happens after. There are also [three tests](https://github.com/ReactionMechanismGenerator/RMG-Py/blob/master/rmgpy/molecule/resonanceTest.py#L81) that are already in Master which are cases which hit an `AtomTypeError` and ultimately return an empty list.